src/transforms/uglify.js

```diff
@@ -1,18 +1,37 @@
-const {AST_Node, minify} = require('uglify-js');
-const {toEstree} = require('babel-to-estree');
-const types = require('babel-types');
-const walk = require('babylon-walk');
+const {minify} = require('uglify-es');
+const generate = require('babel-generator').default;
 
 module.exports = async function(asset) {
   await asset.parseIfNeeded();
 
-  // Convert to UglifyJS AST
-  var ast = AST_Node.from_mozilla_ast(toEstree(asset.ast, asset.contents));
-  var result = minify(ast, {
-    toplevel: true
-  });
+  // Convert AST into JS
+  let code = asset.isAstDirty
+    ? generate(asset.ast).code
+    : asset.outputCode || asset.contents;
 
-  // Uglify did our code generation for us, so remove the old AST
+  let options = {
+    warnings: true,
+    mangle: {
+      toplevel: true
+    },
+    compress: {
+      drop_console: true
+    }
+  };
+
+  let result = minify(code, options);
+
+  if (result.error) throw result.error;
+
+  // Log all warnings
+  if (result.warnings) {
+    result.warnings.forEach(warning => {
+      // TODO: warn this using the logger
```

Comment:
Would probably be better to use the logger to send warnings to the console, not really sure how to address the logger from here though

Reply:
hmm yeah, the logger isn't really accessible from here, it lives only in the main process right now, not workers.